src/components/Search/index.tsx

```diff
@@ -123,9 +132,34 @@ function Search({queryJSON}: SearchProps) {
         }
 
         SearchActions.search({queryJSON, offset});
-        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
     }, [isOffline, offset, queryJSON]);
 
+    const searchTriggeredRef = useRef(false);
```

Comment:
Request: can we move this animation/scrolling logic into a custom hook so we can keep the logic in the Search component small?